osu.Game.Rulesets.Osu/Mods/OsuModBubbles.cs

```diff
@@ -0,0 +1,255 @@
+
+                DrawableOsuHitObject drawableOsuHitObject = lastJudgedHitobject;
+
+                switch (result.HitObject)
+                {
+                    case Slider:
+                    case SpinnerTick:
+                        break;
+
+                    default:
+                        addBubble();
+                        break;
+                }
+
+                void addBubble()
+                {
+                    BubbleDrawable bubble = bubblePool.Get();
+                    bubble.Info = new BubbleInfo
+                    {
+                        InitialSize = new Vector2(bubbleRadius),
+                        MaxSize = maxSize,
+                        Position = getPosition(),
+                        FadeTime = bubbleFade,
+                        Colour = drawableOsuHitObject.AccentColour.Value,
+                        IsHit = drawableOsuHitObject.IsHit,
+                    };
+                    bubbleContainer.Add(bubble);
+                }
+
+                Vector2 getPosition()
+                {
+                    switch (drawableOsuHitObject)
+                    {
+                        // SliderHeads are derived from HitCircles,
+                        // so we must handle them before to avoid them using the wrong positioning logic
+                        case DrawableSliderHead:
+                            return osuHitObject.Position;
+
+                        // Using hitobject position will cause issues with HitCircle placement due to stack leniency.
+                        case DrawableHitCircle:
+                            return drawableOsuHitObject.Position;
+
+                        default:
+                            return osuHitObject.Position;
+                    }
+                }
+            };
+
+            scoreProcessor.JudgementReverted += _ =>
+            {
+                bubbleContainer.LastOrDefault()?.FinishTransforms();
+                bubbleContainer.LastOrDefault()?.Expire();
+            };
+        }
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
+        {
+            // Multiplying by 2 results in an initial size that is too large, hence 1.90 has been chosen
+            // Also avoids the HitObject bleeding around the edges of the bubble drawable at minimum size
+            bubbleRadius = (float)(drawableRuleset.Beatmap.HitObjects.OfType<HitCircle>().First().Radius * 1.90f);
+            bubbleFade = drawableRuleset.Beatmap.HitObjects.OfType<HitCircle>().First().TimePreempt * 2;
+
+            // We want to hide the judgements since they are obscured by the BubbleDrawable (due to layering)
+            drawableRuleset.Playfield.DisplayJudgements.Value = false;
+
+            bubbleContainer = drawableRuleset.CreatePlayfieldAdjustmentContainer();
+
+            drawableRuleset.KeyBindingInputManager.Add(bubbleContainer);
```

Comment:
I'm wondering if you can use `drawableRuleset.Overlays` instead. It would feel like a much more correct place if so (and may also be appropriate for flashlight mods if it works here).

Reply:
yep it works there, and used to be there, only changed for FL compatibility purposes, now that that is changed there should be no reason for it to remain here